### ISSUE TITLE
Question regarding the gradient accumulation

### ISSUE THREAD

**Issue by lehduong** [kxhit/zero123-hf#7]
Hi, thanks for your implementation. 

I noticed you accumulated gradients of two models with two different context managers ([here](https://github.com/kxhit/zero123-hf/blob/8935660dff9048009c78ed3009042764b5c5b2c2/train_zero1to3.py#L834)). Could you let me know if you verified your implementation with gradient accumulation step different than 1? Apparently, this approach can be erroneous according to [this](https://github.com/huggingface/accelerate/issues/668#issuecomment-1574438411) and followed-up comments. I believe the newer version of hf's accelerate has already allowed the context manager to receive multiple models as in [here](https://github.com/huggingface/diffusers/issues/4046#issuecomment-1686548737).

**Comment by kxhit** [comment]
Hi @lehduong thanks for pointing this out! Yes, in Zero123, the gradient accumulation is 1 so this doesn't matter. And I know people are trying to fix it for more than one model. According to your link, the correct way to do it for multiple models is 
`with accelerator.accumulate(model1, model2)`? Would be very happy if you want to PR! Thank you!

**Comment by lehduong** [comment]
Yes, you only need to change this [line](https://github.com/kxhit/zero123-hf/blob/8935660dff9048009c78ed3009042764b5c5b2c2/train_zero1to3.py#L834) to `with accelerator.accumulate(unet, cc_projection)` and use `accelerate >= 0.23` (not sure if it is the earliest accelerate version that supports this feature but I'm using it). I did a quick experiment and I attached the image below to compare the training (epoch) loss of two accumulation approaches. I used the resolution of 512, batch size of 24, and the gradient_accumulation_steps is set to 8 (effective batch size is 1536 as in the original implementation).

![Screenshot 2023-10-20 at 2 17 19 PM](https://github.com/kxhit/zero123-hf/assets/27000711/a6230b19-768a-474b-a2cc-92173410f09a)

**Comment by kxhit** [comment]
Thanks a lot for your contribution! Just pushed a fix, please reopen the issue if anything needs further change!

**Comment by cfeng16** [comment]
Hi @lehduong , can I ask do you use gradient accumulation for multiple models in distributed training setting (say multiple GPUs)?

**Comment by lehduong** [comment]
I did, but I trained the model on a single machine (8gpus).
